pkg/builder/csolution/builder.go

```diff
@@ -735,3 +690,95 @@ func (b CSolutionBuilder) hasRebuildNode(idxFilePath string) (bool, error) {
 
 	return false, nil
 }
+
+func (b CSolutionBuilder) Clean() (err error) {
+	// Get list of cleanable contexts
+	cleanableContexts, err := b.getContextsToClean()
+	if err != nil {
+		return err
+	}
+
+	outputDir := b.Options.Output
+
+	var tmpDir string
+	if !b.Options.UseCbuild2CMake {
+		// Use default path when --cbuildgen option is used
+		tmpDir = filepath.Join(filepath.Dir(b.InputFile), outputDir, "tmp")
+	} else {
+		tmpDir, err = utils.GetTmpDir(b.InputFile, outputDir)
+		if err != nil {
+			return err
```

Comment:
I believe the non existence of the `tmp` directory should not be a reason to return error.